### PR TITLE
Enable to highlight CJK mention key

### PR DESCRIPTION
**Pull request by kaakaa**
#### Summary
https://github.com/mattermost/mattermost-server/pull/8904#issuecomment-397086557
> 1. The Chinese characters triggering mentions are not highlighting in the post as English characters do:

With this PR, trigger words with CJK character will be highlighted.

#### Ticket Link
[\[Help Wanted\] \[MM\-10193\] Words that trigger mentions does not support Chinese · Issue \#8642 · mattermost/mattermost\-server](https://github.com/mattermost/mattermost-server/issues/8642)
[\[MM\-10193\] Words that trigger mentions does not support Chinese \- Mattermost](https://mattermost.atlassian.net/browse/MM-10193)

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit and component tests passed
- [x] Added or updated unit tests (required for all new features)
- [x] Has server changes (please link)
  - https://github.com/mattermost/mattermost-server/pull/10262

### CODE REVIEW

**Comment by jasonblais** [comment]
Same as https://github.com/mattermost/mattermost-server/pull/10262 - added for QA review. Also, added dev review label in case @hmhealey you want to give an initial review of the direction for this PR.

File: utils/text_formatting.jsx [hunk 1]
```diff
@@ -319,8 +319,13 @@ function highlightCurrentMentions(text, tokens, mentionKeys = []) {
             flags += 'i';
         }
 
-        const pattern = new RegExp(`(^|\\W)(${escapeRegex(mention.key)})(\\b|_+\\b)`, flags);
-
+        let pattern;
+        if (cjkPattern.test(mention.key)) {
+            // In the case of CJK mention key, even if there no delimiters (such as spaces) at both ends of a word, it is recognized as a mention key
+            pattern = new RegExp(`(^|.)(${escapeRegex(mention.key)})($|.)`, flags);
```
Comment (hmhealey) [inline utils/text_formatting.jsx]: For the prefix and suffix groups, you can just use empty brackets like `()` to get the same results. I think the ones that you’re using may not match text right at the start or end of a line in all cases

Comment (kaakaa) [reply utils/text_formatting.jsx]: You're right! I've fix this.

**Comment by lindalumitchell** [comment]
Thanks @kaakaa and @jasonblais! This looks great (I also tested https://github.com/mattermost/mattermost-server/pull/10262). All given examples work as expected, and no issues found when spot-checking other mention functionality. LGTM! 👍

**Comment by hanzei** [comment]
@migbot Please review, when you have the time

File: utils/text_formatting.jsx [hunk 2]
```diff
@@ -326,8 +326,13 @@ function highlightCurrentMentions(text, tokens, mentionKeys = []) {
             flags += 'i';
         }
 
-        const pattern = new RegExp(`(^|\\W)(${escapeRegex(mention.key)})(\\b|_+\\b)`, flags);
-
+        let pattern;
+        if (cjkPattern.test(mention.key)) {
+            // In the case of CJK mention key, even if there no delimiters (such as spaces) at both ends of a word, it is recognized as a mention key
```
Comment (saturninoabril) [inline utils/text_formatting.jsx]: Good to change but non-blocking
```suggestion
            // In the case of CJK mention key, even if there's no delimiters (such as spaces) at both ends of a word, it is recognized as a mention key
```

**Comment by kaakaa** [comment]
Thanks for your review @saturninoabril!
I accepted your suggestion, and rebased. (rebased server-side too https://github.com/mattermost/mattermost-server/pull/10262)

And, I did "force-push" for fixing the author of my new commit because at first I made the commit with my another github account. I didn't make any changes for code. sorry for my mistakes.

**Comment by saturninoabril** [comment]
Thank you @kaakaa and no worries!

**Pull request merged**